notebooks/compute-network-v2.ipynb

```diff
@@ -0,0 +1,464 @@
+    "data_sub = data.loc[r, ['dt', 'bw', 'tw', 'twcc', 'dx', 'n', 'ncc', 'cs', 's0']].sort_index()\n",
+    "qlat_sub = qlats.loc[r].sort_index()\n",
+    "  \n",
+    "# compute the network routing, calculate (flow, depth, and velocity)    \n",
+    "results.append(mc_reach.compute_network(\n",
+    "    nts, reach, subnets[tw], data_sub.index.values, data_sub.columns.values, data_sub.values, qlat_sub.values\n",
+    "    )\n",
+    ")\n",
+    "\n",
+    "# create a multi-index DataFrame with flow, depth, and velocity simulations\n",
+    "fdv_columns = pd.MultiIndex.from_product([range(nts), ['q', 'v', 'd']])\n",
+    "flowveldepth = pd.concat([pd.DataFrame(d, index=i, columns=fdv_columns) for i, d in results], copy=False)\n",
+    "flowveldepth = flowveldepth.sort_index()\n",
+    "\n",
+    "print(\"Routing computations complete!\")\n",
+    "print(\"... in %s minutes.\" % ((time.time() - start_time)/60))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Extract simulated flow data for each segment in the Mississippi River Basin subnetwork"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# grab simulated flows for all segments in the network\n",
+    "# [i, j, k]:\n",
+    "# i = index values for each segment\n",
+    "# j = all times\n",
+    "# k = the parameter of interest\n",
+    "network_flows = flowveldepth.loc[:, (slice(None), 'q')]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Create a GeoDataFrame for viz purposes"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# create a GeoDataFrame of points for each link in the stream network\n",
+    "dat_geo = nhd_io.read(network_data[\"geo_file_path\"])\n",
+    "dat_geo = dat_geo.set_index(network_data[\"key_col\"])\n",
+    "\n",
+    "if \"mask_file_path\" in network_data:\n",
+    "        data_mask = nhd_io.read_mask(\n",
+    "            network_data[\"mask_file_path\"],\n",
+    "            layer_string=network_data[\"mask_layer_string\"],\n",
+    "        )\n",
+    "        dat_geo = dat_geo.filter(data_mask.iloc[:, network_data[\"mask_key\"]], axis=0)\n",
+    "\n",
+    "# create a GeoData Frame        \n",
+    "gdf = geopandas.GeoDataFrame(dat_geo, geometry=geopandas.points_from_xy(dat_geo.lon, dat_geo.lat))\n",
+    "\n",
+    "# clip the GeoDataFrame to include ONLY segments in the specified river network\n",
+    "network_segs = subreachable[tw]\n",
+    "network_gdf = gdf.loc[gdf.index.isin(network_segs)]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Create an animation showing flow changes through space and time"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import gif\n",
```

Comment:
Will need:
```
if ENV_IS_CL:
    subprocess.run(["pip", "install", "gif"])
```

Reply:
all ! statements were changed to subprocess.run. Thanks you you and @jhrehanoaa for doing this in the first cell for me